.github/actions/create-branch-tag/action.yml

```diff
@@ -0,0 +1,46 @@
+name: Create a branch and a tag
+description: Create a branch and a tag from the version provided
+
+inputs:
+  tag-name:
```

Comment:
I added a step to check the format of the tag input.
Instead of letting this composite action figure out whether we are in a pre-release context (by checking if "test" is present in the tag), it is now up to the parent actions to pass that info. That way, we can check if the tag respects the expected format (release or test) here.
We trim the leading and ending whitespace before checking, and the trimmed value is set as an output to be reused by parent actions.

If the format is not OK, we stop here, and the workflow will fail.